naga/src/back/spv/writer.rs

```diff
@@ -823,6 +823,15 @@ impl Writer {
                 if bits == 64 {
                     self.capabilities_used.insert(spirv::Capability::Float64);
                 }
+                if bits == 16 {
+                    self.capabilities_used.insert(spirv::Capability::Float16);
+                    self.capabilities_used
+                        .insert(spirv::Capability::StorageBuffer16BitAccess);
+                    self.capabilities_used
+                        .insert(spirv::Capability::UniformAndStorageBuffer16BitAccess);
```

Comment:
@teoxoy not actually sure about this now.
https://github.com/gpuweb/gpuweb/issues/2512#issuecomment-1034413409
Seems some of these are optional. If you enable them all you lose most devices on Vulkan.

Reply:
What ended up in the spec requires all of the Vulkan features. Some more context from the follow-up meetings:
https://github.com/gpuweb/gpuweb/issues/2512#issuecomment-1041207899
https://github.com/gpuweb/gpuweb/issues/2512#issuecomment-1057224605